connector_bots_stock_integration/supplier_stock.py

```
@@ -0,0 +1,187 @@
+from openerp.addons.connector_bots.backend import bots
+from openerp.addons.connector_bots.connector import get_environment, add_checkpoint
+import openerp.addons.decimal_precision as dp
+import json
+import traceback
+from datetime import datetime
+
+from psycopg2 import OperationalError
+_logger = logging.getLogger(__name__)
+
+@bots
+class BotsStockImport(ImportSynchronizer):
+    _model_name = ['bots.product']
+
+    def import_supplier_stock(self, new_cr=True):
+        """
+        Import the picking confirmation from Bots
+        """
+        self.backend_adapter.get_supplier_stock(new_cr=new_cr)
+
+@bots
+class StockAdapter(BotsCRUDAdapter):
+    _model_name = 'bots.product'
+    
+    def _read_csv(self, record_file):
+        """ Returns a CSV-parsed iterator of all empty lines in the file
+
+        :throws csv.Error: if an error is detected during CSV parsing
+        :throws UnicodeDecodeError: if ``options.encoding`` is incorrect
+        """
+        cr = pooler.get_db(self.session.cr.dbname).cursor()
+        file_obj = self.session.pool.get('bots.file')
+        file = file_obj.browse(cr, SUPERUSER_ID, record_file)
+        fd = open(file.full_path, "rb")
+        csv_iterator = csv.reader(
+            fd,
+            quotechar=str('"'),
+            delimiter=str(','))
+        csv_nonempty = itertools.ifilter(None, csv_iterator)
+        encoding = 'utf-8'
+        ret = itertools.imap(
+            lambda row: [item.decode(encoding) for item in row],
+            csv_nonempty)
+        cr.commit()
+        cr.close()
+        return ret
+        
+    def _check_product(self, barcode=False, sku=False):
+        cr = pooler.get_db(self.session.cr.dbname).cursor()
+        if barcode and sku:
+            prodids = self.session.pool.get('product.product').search(cr, SUPERUSER_ID, [('magento_barcode','=',barcode), ('magento_supplier_sku','=',sku)])
+            if prodids:
```

This should return the product id if 1 is found, but if we have more than 1 it should throw an error as we wont know which is the correct one to update.